From 2950cb121e2e2e2dbfb4c02b46f9344427101cb9 Mon Sep 17 00:00:00 2001
Message-ID: <2950cb121e2e2e2dbfb4c02b46f9344427101cb9.1688558232.git.stefan@agner.ch>
From: David Rheinsberg <david@readahead.eu>
Date: Wed, 5 Jul 2023 09:41:11 +0200
Subject: [PATCH] launch/config: avoid expat.h in header

To reduce exposure of expat, avoid it in config.h. The only definition
we use can be easily forward-declared, and thus we can completely
contain expat in config.c.

Signed-off-by: David Rheinsberg <david@readahead.eu>
---
 src/launch/config.h | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/launch/config.h b/src/launch/config.h
index c36819d..881974a 100644
--- a/src/launch/config.h
+++ b/src/launch/config.h
@@ -6,9 +6,10 @@
 
 #include <c-list.h>
 #include <c-stdaux.h>
-#include <expat.h>
 #include <stdlib.h>
 
+struct XML_ParserStruct;
+
 typedef struct ConfigPath ConfigPath;
 typedef struct ConfigNode ConfigNode;
 typedef struct ConfigParser ConfigParser;
@@ -218,7 +219,7 @@ struct ConfigRoot {
         }
 
 struct ConfigParser {
-        XML_Parser xml;
+        struct XML_ParserStruct *xml;
 
         struct ConfigState {
                 NSSCache *nss;
-- 
2.41.0

